tests/functional/src/main/java/org/mule/functional/classloading/isolation/classification/ClassPathClassifierContext.java

```diff
@@ -0,0 +1,232 @@
+    private final Predicate<MavenArtifact> exclusions;
+    private final Set<String> extraBootPackages;
+
+    protected final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    /**
+     * Creates a context used for doing the classification of the class path.
+     *
+     * @param testClass the test {@link Class} being tested
+     * @param classPathURLs the whole set of {@link URL}s that were loaded by IDE/Maven Surefire plugin when running the test
+     * @param dependenciesGraph the maven dependencies graph for the artifact that the test belongs to
+     * @param mavenMultiModuleArtifactMapping a mapper to get multi-module folder for artifactIds
+     */
+    public ClassPathClassifierContext(final Class<?> testClass, final List<URL> classPathURLs, final DependenciesGraph dependenciesGraph, final MavenMultiModuleArtifactMapping mavenMultiModuleArtifactMapping) throws IOException
+    {
+        this.testClass = testClass;
+        this.classPathURLs = classPathURLs;
+        this.dependenciesGraph = dependenciesGraph;
+        this.mavenMultiModuleArtifactMapping = mavenMultiModuleArtifactMapping;
+
+        Properties excludedProperties = getExcludedProperties();
+        this.exclusions = createExclusionsPredicate(testClass, excludedProperties);
+        this.extraBootPackages = getExtraBootPackages(testClass, excludedProperties);
+    }
+
+    /**
+     * @return the {@link Class} for the test that is going to be executed
+     */
+    public Class<?> getTestClass()
+    {
+        return testClass;
+    }
+
+    /**
+     * @return a {@link List<URL>} of URLs for the classpath provided by JUnit (it is the complete list of URLs)
+     */
+    public List<URL> getClassPathURLs()
+    {
+        return classPathURLs;
+    }
+
+    /**
+     * @return a {@link DependenciesGraph} for the given artifact tested.
+     */
+    public DependenciesGraph getDependencyGraph()
+    {
+        return dependenciesGraph;
+    }
+
+    /**
+     * @return {@link MavenMultiModuleArtifactMapping} mapper for artifactIds and multi-module folders.
+     */
+    public MavenMultiModuleArtifactMapping getMavenMultiModuleArtifactMapping()
+    {
+        return mavenMultiModuleArtifactMapping;
+    }
+
+    /**
+     * @return {@link Predicate<MavenArtifact>} to be used to exclude artifacts from being added to application {@link ClassLoader} due to
+     * they are going to be in container {@link ClassLoader}.
+     */
+    public Predicate<MavenArtifact> getExclusions()
+    {
+        return exclusions;
+    }
+
+    /**
+     * @return {@link Set<String>} containing the extra boot packages defined to be appended to the container in addition to the pre-defined ones.
+     */
+    public Set<String> getExtraBootPackages()
+    {
+        return extraBootPackages;
+    }
+
+    /**
+     * The list of exclusion GAT to be excluded from application/plugin class loaders due to these are supposed to
```

Comment:
GroupId, ArtifactId, Type, from maven glossary. Some plugins use this abbreviations for those key words. I reverted this to have the complete words.